astrosat_users/views/views_customers.py

```diff
@@ -13,12 +13,26 @@
 from astrosat_users.serializers import CustomerSerializer, CustomerUserSerializer
 
 
+class HasPendingCustomer(BasePermission):
+    """
+    Only a user w/ a pending_customer can create a customer
```

Comment:
The class is called `HasPendingCustomer` because it checks if the `user` has the flag `pending_customer`.  

As both you and @Reignable have some concerns about the name of that field I will think about changing it.

The unused arguments cannot be removed.  They form the method signature and this is overwriting the builtin `has_permission` method; it can only do that if the signatures match.

Reply:
I will change the name to `requires_customer_registration_completion` as suggested.  It isn't precisely accurate from a backend point of view, but it does get the point of this field across for other developers.